lib/Doctrine/ORM/Tools/SchemaTool.php

```diff
@@ -737,33 +737,35 @@ private function gatherRelationJoinColumns(
 
         $compositeName = $theJoinTable->getName() . '.' . implode('', $localColumns);
 
-        if (isset($addedFks[$compositeName])
-            && ($foreignTableName !== $addedFks[$compositeName]['foreignTableName']
-            || 0 < count(array_diff($foreignColumns, $addedFks[$compositeName]['foreignColumns'])))
-        ) {
-            foreach ($theJoinTable->getForeignKeys() as $fkName => $key) {
-                if (count(array_diff($key->getLocalColumns(), $localColumns)) === 0
-                    && (($key->getForeignTableName() !== $foreignTableName)
-                    || 0 < count(array_diff($key->getForeignColumns(), $foreignColumns)))
-                ) {
-                    $theJoinTable->removeForeignKey($fkName);
-                    break;
+        if ($this->platform->supportsForeignKeyConstraints()) {
+            if (isset($addedFks[$compositeName])
+                && ($foreignTableName !== $addedFks[$compositeName]['foreignTableName']
+                    || 0 < count(array_diff($foreignColumns, $addedFks[$compositeName]['foreignColumns'])))
+            ) {
+                foreach ($theJoinTable->getForeignKeys() as $fkName => $key) {
+                    if (count(array_diff($key->getLocalColumns(), $localColumns)) === 0
+                        && (($key->getForeignTableName() !== $foreignTableName)
+                            || 0 < count(array_diff($key->getForeignColumns(), $foreignColumns)))
+                    ) {
+                        $theJoinTable->removeForeignKey($fkName);
+                        break;
+                    }
                 }
-            }
 
-            $blacklistedFks[$compositeName] = true;
-        } elseif (! isset($blacklistedFks[$compositeName])) {
-            $addedFks[$compositeName] = [
-                'foreignTableName' => $foreignTableName,
-                'foreignColumns'   => $foreignColumns,
-            ];
-
-            $theJoinTable->addForeignKeyConstraint(
-                $foreignTableName,
-                $localColumns,
-                $foreignColumns,
-                $fkOptions
-            );
+                $blacklistedFks[$compositeName] = true;
+            } elseif (! isset($blacklistedFks[$compositeName])) {
+                $addedFks[$compositeName] = [
+                    'foreignTableName' => $foreignTableName,
+                    'foreignColumns' => $foreignColumns,
+                ];
+
+                $theJoinTable->addForeignKeyConstraint(
+                    $foreignTableName,
+                    $localColumns,
+                    $foreignColumns,
+                    $fkOptions
+                );
+            }
```

Comment:
This portion of the code was already a bit complex and adding an if here just increases the cyclomatic complexity. Would you please try to use some early returns to make this a bit more readable and simpler? Like:

```php
if (! $this->platform->supportsForeignKeyConstraints()) {
    return;
}

if (isset($addedFks[$compositeName]) && ($foreignTableName !== $addedFks[$compositeName]['foreignTableName'] || array_diff($foreignColumns, $addedFks[$compositeName]['foreignColumns']) !== [])) { // expression simplified as well
    // (...)
}

if (isset($blacklistedFks[$compositeName])) {
    return;
}

$addedFks[$compositeName] = ['foreignTableName' => $foreignTableName, 'foreignColumns' => $foreignColumns];
$theJoinTable->addForeignKeyConstraint($foreignTableName, $localColumns, $foreignColumns, $fkOptions);
```

Reply:
I was believe, it should be in refactoring stage unrelated to any issue. But ok, will do it.